cmd/orch-tester/orch_tester.go

```diff
@@ -68,6 +70,134 @@ type broadcasterConfig struct {
 	maxPricePerUnit *int
 }
 
+type broadcasterMetrics struct {
+	// Distribution metrics grouped by metric name
+	lastSum   map[string]float64
+	lastCount map[string]int
+	rateSum   map[string]float64
+	rateCount map[string]int
+
+	// Error count metrics by grouped by error_code tag
+	lastErrs map[string]int
+	incErrs  map[string]int
+
+	mu sync.Mutex
+}
+
+func newBroadcasterMetrics() *broadcasterMetrics {
+	return &broadcasterMetrics{
+		lastSum:   map[string]float64{},
+		lastCount: map[string]int{},
+		rateSum:   map[string]float64{},
+		rateCount: map[string]int{},
+		lastErrs:  map[string]int{},
+		incErrs:   map[string]int{},
+		mu:        sync.Mutex{},
+	}
+}
+
+func (bm *broadcasterMetrics) reset() {
+	bm.mu.Lock()
+	defer bm.mu.Unlock()
+
+	bm.rateSum = map[string]float64{}
+	bm.rateCount = map[string]int{}
+	bm.incErrs = map[string]int{}
+}
+
+func (bm broadcasterMetrics) ExportView(viewData *view.Data) {
+	bm.mu.Lock()
+	defer bm.mu.Unlock()
+
+	name := viewData.View.Name
+	rows := viewData.Rows
+	if len(rows) <= 0 {
+		return
+	}
+
+	bm.handleDistributionMetrics(rows, name)
+	bm.handleErrorCountMetrics(rows, name)
+}
+
+func (bm broadcasterMetrics) handleDistributionMetrics(rows []*view.Row, name string) {
+	supportedMetrics := map[string]bool{
+		"source_segment_duration_seconds":   true,
+		"transcode_overall_latency_seconds": true,
+		"upload_time_seconds":               true,
+		"download_time_seconds":             true,
+	}
+	if !supportedMetrics[name] {
+		return
+	}
+
+	// Sum metrics with different tags
+	var sum float64
+	var count int
+	for _, r := range rows {
+		d, ok := r.Data.(*view.DistributionData)
+		if ok {
+			sum += d.Sum()
+			count += int(d.Count)
+		}
+	}
+
+	// Get the rate between the last measurement
+	rateSum := sum - bm.lastSum[name]
+	rateCount := count - bm.lastCount[name]
+
+	bm.rateSum[name] += rateSum
+	bm.rateCount[name] += rateCount
+	bm.lastSum[name] = sum
+	bm.lastCount[name] = count
```

Comment:
I'm wondering if we really need the `lastSum/Count` values here, I might not have picked up some of the logic. Would this generate the same results?
```suggestion
	bm.rateSum[name] = sum
	bm.rateCount[name] = count
```

Reply:
No, actually not. It's tricky, but census always gives the sum of all the previous metrics. So, in result, if we use these values as you mentioned, for the given O, we would take the values for all the past O measurements.

In other words, your code would correspond to the prometheus query like this one:
```
sum(livepeer_metricname_sum)/sum(livepeer_metricname_count)
```

While we actually want to scope it to one orchestrator only. So, a query like this (`1m` might not be exactly accurate, but that's what worked):

```
sum(rate(livepeer_metricname_sum[1m]))/sum(rate(livepeer_metricname_count[1m]))
```